pkg/manager/member/tidb_member_manager.go

```diff
@@ -242,9 +243,44 @@ func (m *tidbMemberManager) syncTiDBStatefulSetForTidbCluster(tc *v1alpha1.TidbC
 		}
 	}
 
+	// when contains a not exist mount, instead of update STS -> rolling update(delete pod + keep failed to create pod)...
+	// we skip update the STS and add event in tc to let user learn this and fix the spec of tc.
```

Comment:
```suggestion
	// If the StatefulSet spec contains a volume mount that does not have a matched volume,
	// do not update the StatefulSet, otherwise, it will trigger a rolling update and the
	// rolling update will hang because the Pod cannot be created.
	// Emit event and return error here to let the user be aware of this and fix it in the spec
	// TODO: Forbid updating `storageVolumes` in the webhook
```

Reply:
Please also update the code for PD, TiKV, TiCDC, TiFlash, and ngMonitor.

Reply:
Please also update UT.